### PR TITLE
Catch common target initialization error

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: cocos-creator/2d-tasks#

Changes:
 * This should fix that engine would hang up if there is an error thrown during initialization of animation common target.

### CODE REVIEW

File: cocos/core/animation/animation-state.ts [hunk 1]
```diff
@@ -394,9 +399,14 @@ export class AnimationState extends Playable {
...
-                target: new BufferedTarget(root, commonTarget.modifiers, commonTarget.valueAdapter),
-                changed: false,
+             try {
+                return {
+                    target: new BufferedTarget(root, commonTarget.modifiers, commonTarget.valueAdapter),
+                    changed: false,
+                };
+            } catch (err) {
+                error(err);
+                return null;
```
Comment (pandamicro) [inline cocos/core/animation/animation-state.ts]: try catch 逻辑限定在编辑器下，运行时不要有任何 try catch 逻辑

File: cocos/core/animation/animation-state.ts [hunk 2]
```diff
@@ -411,12 +421,21 @@ export class AnimationState extends Playable {
...
+                rootTarget = root;
+            } else {
+                const commonTargetStatus = this._commonTargetStatuses[propertyCurve.commonTarget];
+                if (!commonTargetStatus) {
+                    continue;
+                }
+                rootTarget = commonTargetStatus.target.peek();
+            }
+
             try {
```
Comment (pandamicro) [inline cocos/core/animation/animation-state.ts]: 同上

**Review by pandamicro — CHANGES_REQUESTED** [review]

File: cocos/core/animation/animation-state.ts (hunk 1 above)
Comment (shrinktofit) [reply cocos/core/animation/animation-state.ts]: 已修改

File: cocos/core/animation/animation-state.ts (hunk 2 above)
Comment (shrinktofit) [reply cocos/core/animation/animation-state.ts]: 已修改

**Comment by pandamicro** [comment]
ci 挂了哦

File: cocos/core/animation/bound-target.ts [hunk 3]
```diff
@@ -5,101 +5,116 @@
...
+    getValue (): any;
+}
+
+export interface IBufferedTarget extends IBoundTarget {
+    peek(): any;
+    pull(): void;
+    push(): void;
+}
+
+export function BoundTarget (target: any, modifiers: TargetPath[], valueAdapter?: IValueProxyFactory): null | IBoundTarget {
```
Comment (pandamicro) [inline cocos/core/animation/bound-target.ts]: 函数命名有问题，BoundTarget 会让人误认为是 class，如果是这种工具函数，应该使用 createBoundTarget 之类的名字，并且保证首字母小写

File: cocos/core/animation/bound-target.ts [hunk 4]
```diff
@@ -5,101 +5,116 @@
...
-    constructor (target: any, modifiers: TargetPath[], valueAdapter?: IValueProxyFactory) {
-        super(target, modifiers, valueAdapter);
-        const value = this.getValue();
-        const copyable = getBuiltinCopy(value);
-        if (!copyable) {
-            throw new Error(`Value is not copyable!`);
-        }
-        this._buffer = copyable.createBuffer();
-        this._copy = copyable.copy;
+export function BufferedTarget (target: any, modifiers: TargetPath[], valueAdapter?: IValueProxyFactory): null | IBufferedTarget {
```
Comment (pandamicro) [inline cocos/core/animation/bound-target.ts]: 同上

**Review by pandamicro — CHANGES_REQUESTED** [review]

**Review by pandamicro — APPROVED** [review]

**Pull request merged**